File: PayTR Direkt API/PayTR Mağaza Durum Sorgulama/English/Sample PYTHON/status_query_example.py
# Python 3.6+

import base64
import hmac
import hashlib
import requests
import json
import random


merchant_id = 'XXXXXX'
merchant_key = b'XXXXXXXXXXXXXXXXXX'
merchant_salt = 'XXXXXXXXXXXXXXXXXX'
merchant_oid = ''





hash_str = merchant_id + merchant_oid + merchant_salt
paytr_token = base64.b64encode(hmac.new(merchant_key, hash_str.encode(), hashlib.sha256).digest())

params = {
    'merchant_id': merchant_id,
    'merchant_oid': merchant_oid,
    'paytr_token': paytr_token
}

result = requests.post('https://www.paytr.com/odeme/durum-sorgu', params)
res = json.loads(result.text)

if res['status'] == 'success':
    print(res['payment_amount'] + res['currency'])
    print(res['payment_total'] + res['currency'])

    for return_success in res['returns']:
      print(return_success)

else:
    print(res['err_no'] + ' ' + res['err_msg'])
    
    

File: PayTR Direkt API/PayTR Mağaza Durum Sorgulama/English/Sample PHP/status_query_example.php
<?php
    $merchant_id = "XXX";
    $merchant_key = "XXX";
    $merchant_salt = "XXX";
    $merchant_oid = "XXX";

    $paytr_token = base64_encode(hash_hmac('sha256', $merchant_id . $merchant_oid . $merchant_salt, $merchant_key, true));

    $post_vals = array('merchant_id' => $merchant_id,
        'merchant_oid' => $merchant_oid,
        'paytr_token' => $paytr_token);

    $ch = curl_init();
    curl_setopt($ch, CURLOPT_URL, "https://www.paytr.com/odeme/durum-sorgu");
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
    curl_setopt($ch, CURLOPT_POST, 1);
    curl_setopt($ch, CURLOPT_POSTFIELDS, $post_vals);
    curl_setopt($ch, CURLOPT_FRESH_CONNECT, true);
    curl_setopt($ch, CURLOPT_TIMEOUT, 90);
    curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 90);
	

	
    $result = @curl_exec($ch);

    if (curl_errno($ch)) {
        echo curl_error($ch);
        curl_close($ch);
        exit;
    }
    curl_close($ch);

    $result = json_decode($result, 1);

    if ($result[status] != 'success') {
        echo $result[err_no] . " - " . $result[err_msg];
        exit;
    }


    echo $result[payment_amount] . " " . $result[currency] . "<br>";


    echo $result[payment_total] . " " . $result[currency] . "<br>";


    foreach ($result[returns] AS $return_success)
        print_r($return_success);
?>